spring-kafka/src/test/java/org/springframework/kafka/config/KafkaStreamsCustomizerTests.java

```diff
@@ -168,7 +172,26 @@ public KafkaStreamsConfiguration kStreamsConfigs() {
 		}
 
 		private KafkaStreamsCustomizer customizer() {
-			return kafkaStreams -> kafkaStreams.setStateListener(STATE_LISTENER);
+			return new KafkaStreamsCustomizer() {
+				@Override
+				public KafkaStreams initKafkaStreams(
+						final Topology topology,
+						final Properties properties,
+						final KafkaClientSupplier clientSupplier
+				) {
+					ksInitialized.set(true);
+					return KafkaStreamsCustomizer.super.initKafkaStreams(
+							topology,
+							properties,
+							clientSupplier
+					);
+				}
+
+				@Override
+				public void customize(final KafkaStreams kafkaStreams) {
+					kafkaStreams.setStateListener(STATE_LISTENER);
```

Comment:
Why are we setting this? Can we also add an assertion?

Reply:
this was already there in the initial test (see previous line 171) I just converted it from a lambda to an anonymous class so I can implement multiple methods. I believe the assertions are also already covered:
```
		assertThat(STATE_LISTENER.getCurrentState()).isEqualTo(state);
```